workflows/temporal/temporal.go

```diff
@@ -106,42 +113,52 @@ func (c *TemporalWF) Terminate(ctx context.Context, req *workflows.WorkflowRefer
 	return nil
 }
 
-func (c *TemporalWF) Get(ctx context.Context, req *workflows.WorkflowReference) (*workflows.StateResponse, error) {
+func (c *TemporalWF) Get(ctx context.Context, req *workflows.GetRequest) (*workflows.StateResponse, error) {
 	c.logger.Debugf("getting workflow data")
 	resp, err := c.client.DescribeWorkflowExecution(ctx, req.InstanceID, "")
 	if err != nil {
 		return nil, err
 	}
+
+	var createdAtTime time.Time
+	if resp.WorkflowExecutionInfo.StartTime != nil {
+		createdAtTime = *resp.WorkflowExecutionInfo.StartTime
+	}
+
 	// Build the output struct
 	outputStruct := workflows.StateResponse{
-		WFInfo:    workflows.WorkflowReference{InstanceID: req.InstanceID},
-		StartTime: resp.WorkflowExecutionInfo.StartTime.Format(time.RFC3339),
-		Metadata: map[string]string{
-			"task_queue": resp.WorkflowExecutionInfo.GetTaskQueue(),
-			"status":     lookupStatus(resp.WorkflowExecutionInfo.Status),
+		Workflow: &workflows.WorkflowState{
+			InstanceID:    req.InstanceID,
+			CreatedAt:     createdAtTime,
+			LastUpdatedAt: createdAtTime,
+			RuntimeStatus: lookupStatus(resp.WorkflowExecutionInfo.Status),
+			Properties: map[string]string{
+				"task_queue": resp.WorkflowExecutionInfo.GetTaskQueue(),
+			},
 		},
 	}
 
 	return &outputStruct, nil
 }
 
 func (c *TemporalWF) RaiseEvent(ctx context.Context, req *workflows.RaiseEventRequest) error {
```

Comment:
Even though temporal is not currently implemented, should we still be updating the "LastUpdatedAt" field inside this method?

Reply:
`LastUpdatedAt` only applies to the `GetWorkflow` API. What will normally happen, however, is that someone will raise an event to a Temporal workflow, the Temporal workflow will respond to that event, and then the temporal engine will update the workflow's last updated time automatically. We don't need to do anything for that to happen here. The next time someone calls `GetWorkflow`, the returned workflow state struct will automatically have the updated `LastUpdatedAt` value.